previous/cogs/errors.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from nextcord.ext.commands import (
+    Cog,
+    Context,
+    CommandNotFound,
+    TooManyArguments,
+    BadArgument,
+    MissingRequiredArgument,
+    MissingRole,
+)
+from nextcord import NotFound
+from nextcord.ext.application_checks import ApplicationMissingRole
+
+if TYPE_CHECKING:
+    from nextcord import Interaction
+
+    from ..__main__ import MyBot
+
+
+class Errors(Cog):
+    def __init__(self, bot: MyBot):
+        self.bot = bot
+
+    @Cog.listener()
+    async def on_command_error(self, ctx: Context, error: Exception):
+        error = getattr(error, "original", error)
+
+        if isinstance(error, CommandNotFound):
+            return
+
+        elif isinstance(error, TooManyArguments):
+            await ctx.send("You are giving too many arguments!")
+        elif isinstance(error, BadArgument):
+            await ctx.send(
+                "The library ran into an error attempting to parse your argument."
+            )
+        elif isinstance(error, MissingRequiredArgument):
+            await ctx.send("You're missing a required argument.")
+        # kinda annoying and useless error.
+        elif isinstance(error, NotFound) and "Unknown interaction" in str(error):
+            return
+        elif isinstance(error, MissingRole):
+            role = ctx.guild.get_role(int(error.missing_role))  # type: ignore
+            await ctx.send(f'"{role.name}" is required to use this command.')  # type: ignore
+        else:
```

Comment:
maybe have some NotOwner error?

Reply:
I simply transferred the current error handler, half out of scope even though this entire PR is quite feature/fix creep

Reply:
yeah but worth adding